core/meterer/on_demand_meterer_test.go

```diff
@@ -0,0 +1,33 @@
+package meterer
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+var startTime = time.Date(1971, 8, 15, 0, 0, 0, 0, time.UTC)
+
+func TestMeterDispersal(t *testing.T) {
+	timeSource := func() time.Time { return startTime }
+	meterer := NewOnDemandMeterer(100, 10, timeSource)
+
+	reservation, err := meterer.MeterDispersal(500)
+	require.NoError(t, err)
+	require.NotNil(t, reservation)
+	require.True(t, reservation.OK())
+}
+
+func TestCancelDispersal(t *testing.T) {
+	timeSource := func() time.Time { return startTime }
+	meterer := NewOnDemandMeterer(100, 10, timeSource)
+
+	reservation, err := meterer.MeterDispersal(500)
+	require.NoError(t, err)
+	require.NotNil(t, reservation)
+
+	require.NotPanics(t, func() {
+		meterer.CancelDispersal(reservation)
+	})
```

Comment:
Nit, is there a point to wrapping it like this? If it panics, the test will fail, so I'm not sure why you specifically assert this.

Reply:
True. 3d1c8fd4e9246119f3bfc29918d8df3e717696f6